registration/include/pcl/registration/impl/transformation_estimation_symmetric_point_to_plane_lls.hpp

```diff
@@ -0,0 +1,251 @@
+    return;
+  }
+
+  ConstCloudIterator<PointSource> source_it (cloud_src, indices_src);
+  ConstCloudIterator<PointTarget> target_it (cloud_tgt);
+  estimateRigidTransformation (source_it, target_it, transformation_matrix);  
+}
+
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> inline void
+pcl::registration::TransformationEstimationSymmetricPointToPlaneLLS<PointSource, PointTarget, Scalar>::
+estimateRigidTransformation (const pcl::PointCloud<PointSource> &cloud_src,
+                             const std::vector<int> &indices_src,
+                             const pcl::PointCloud<PointTarget> &cloud_tgt,
+                             const std::vector<int> &indices_tgt,
+                             Matrix4 &transformation_matrix) const
+{
+  size_t nr_points = indices_src.size ();
+  if (indices_tgt.size () != nr_points)
+  {
+    PCL_ERROR ("[pcl::TransformationEstimationSymmetricPointToPlaneLLS::estimateRigidTransformation] Number or points in source (%lu) differs than target (%lu)!\n", indices_src.size (), indices_tgt.size ());
+    return;
+  }
+
+  ConstCloudIterator<PointSource> source_it (cloud_src, indices_src);
+  ConstCloudIterator<PointTarget> target_it (cloud_tgt, indices_tgt);
+  estimateRigidTransformation (source_it, target_it, transformation_matrix);  
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> inline void
+pcl::registration::TransformationEstimationSymmetricPointToPlaneLLS<PointSource, PointTarget, Scalar>::
+estimateRigidTransformation (const pcl::PointCloud<PointSource> &cloud_src,
+                             const pcl::PointCloud<PointTarget> &cloud_tgt,
+                             const pcl::Correspondences &correspondences,
+                             Matrix4 &transformation_matrix) const
+{
+  ConstCloudIterator<PointSource> source_it (cloud_src, correspondences, true);
+  ConstCloudIterator<PointTarget> target_it (cloud_tgt, correspondences, false);
+  estimateRigidTransformation (source_it, target_it, transformation_matrix);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointSource, typename PointTarget, typename Scalar> inline void
+pcl::registration::TransformationEstimationSymmetricPointToPlaneLLS<PointSource, PointTarget, Scalar>::
+constructTransformationMatrix (const double & alpha, const double & beta, const double & gamma,
+                               const double & tx,    const double & ty,   const double & tz,
+                               Matrix4 &transformation_matrix) const
+{
+  // Construct the transformation matrix from rotation and translation 
+  transformation_matrix = Eigen::Matrix<Scalar, 4, 4>::Zero ();
```

Comment:
Will the Eigen3 functions to create transform from euler angles help here?

Reply:
Yup, switched to using those.